### PR TITLE
feat(6530): support for no grow/shrink, and better documentation

### PR DESCRIPTION
**Pull request by justinhelmer**
## Description
Adds ability to reset the default behavior of flex components that grow/shrink (`container`, `layout`, `flex`, `d-flex`).

After diving deeper on https://github.com/vuetifyjs/vuetify/issues/6530, I made the following realizations:
- Following the existing `.grow` and `.shrink` patterns would not be ideal, because they use the `!important` tag and could be applied to anything (not just our flex components)
- There is not a need to separate `grow-0` and `shrink-0`. The current `grow` and `shrink` attributes essentially serve to "reset" the _other_ property, since the default behavior without _any_ additional attributes is to grow and shrink.
- The documentation is unclear and needed some love. It is not clear _why_ one would use `grow` or `shrink`, given the fact that the default behavior already sets `flex-grow: 1` and `flex-shrink: 1`. Therefore, I provided better explanation and examples:

capture
> Please note that the documentation only contains the `en` translation. However I noticed the majority of components don't have a translation, so I didn't worry too much about it.

## Motivation and Context
Resolves https://github.com/vuetifyjs/vuetify/issues/6530

## How Has This Been Tested?
Visually. I actually could not get the docs running locally. It was looking to install a package that did not exist:

```
npm ERR! code E404
npm ERR! 404 Not Found: @vuetify/api-generator@^1.5.1
```

However, I have `vuetify` installed in a project and I tested the changes there.

## Markup:

<details>

```vue
<template>
  <v-container fluid>
    <v-layout row>
      <v-flex
        pa-1
      >
        <v-card
          dark
          color="green darken-3"
        >
          <v-card-text>I'm a Flex that will grow and shrink (default behavior)</v-card-text>
        </v-card>
      </v-flex>
      <v-flex
        pa-1
      >
        <v-card
          dark
          color="green darken-1"
        >
          <v-card-text>I'm a Flex that will grow and shrink (default behavior)</v-card-text>
        </v-card>
      </v-flex>
    </v-layout>
    <v-layout row>
      <v-flex
        pa-1
      >
        <v-card
          dark
          color="green darken-1"
        >
          <v-card-text>I'm a Flex that will grow and shrink (default behavior)</v-card-text>
        </v-card>
      </v-flex>
      <v-flex
        grow
        pa-1
      >
        <v-card
          dark
          color="green lighten-1"
        >
          <v-card-text>I'm a Flex with a lot of content that will NOT shrink when the content is wider than the available width in the row. Try resizing the window</v-card-text>
        </v-card>
      </v-flex>
    </v-layout>
    <v-layout row>
      <v-flex
        pa-1
      >
        <v-card
          dark
          color="green darken-4"
        >
          <v-card-text>I'm a Flex that will grow and shrink (default behavior)</v-card-text>
        </v-card>
      </v-flex>
      <v-flex
        shrink
        pa-1
      >
        <v-card
          dark
          color="green darken-3"
        >
          <v-card-text>I'm a Flex that will NOT grow</v-card-text>
        </v-card>
      </v-flex>
    </v-layout>
    <v-layout>
      <v-flex
        grow-shrink-0
        pa-1
      >
        <v-card
          dark
          color="green"
        >
          <v-card-text>I'm a Flex that will NOT grow OR shrink</v-card-text>
        </v-card>
      </v-flex>
      <v-flex
        grow-shrink-0
        pa-1
      >
        <v-card
          dark
          color="green lighten-1"
        >
          <v-card-text>I'm a Flex that will NOT grow OR shrink</v-card-text>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/docs/src/lang/en/framework/Grid.json [hunk 1]
```diff
@@ -87,6 +87,8 @@
       "alignSelfBaseline": "Apply `align-self: baseline`\n\n[MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/align-self)",
       "grow": "Apply `flex-grow: 1`\n\n[MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-grow)",
       "shrink": "Apply `flex-shrink: 1`\n\n[MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-shrink)",
+      "grow-0": "Apply `flex-grow: 0`\n\n[MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-grow)",
+      "shrink=0": "Apply `flex-shrink: 0`\n\n[MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-shrink)",
```
Comment (sh7dm) [inline packages/docs/src/lang/en/framework/Grid.json]: `shrink-0`

**Comment by sh7dm** [comment]
And `dev` is feature frozen so features to `next`, fixes to `master`

**Comment by johnleider** [comment]
Any grid changes will be to the `next` branch.

File: packages/docs/src/lang/en/framework/Grid.json (hunk 1 above)
Comment (justinhelmer) [reply packages/docs/src/lang/en/framework/Grid.json]: good catch; actually should be `grow-shrink-0`. Updated in https://github.com/vuetifyjs/vuetify/pull/6538

**Comment by justinhelmer** [comment]
Closing this PR in favor of https://github.com/vuetifyjs/vuetify/pull/6538, which has been set up against the `next` branch instead of the `dev` branch.

Since the changes in this PR were originally made against the `dev` branch, the deltas between `dev` and `next` cause merge conflicts. cherry picked the commit on top of `next` to enable a fast-forward.

**Pull request closed without merging**